spec/controllers/hq/users_controller_spec.rb

```diff
@@ -31,4 +31,33 @@ def foobar_action
     expect(response).to render_template 'index'
   end
 
+  describe '#edit and #update' do
+    before :each do
+      @old_user = FactoryGirl.build_stubbed(:user)
+      expect(User).to receive(:find).and_return(@old_user)
+    end    
+
+    specify 'editing renders the edit view' do
+      get :edit, id: @old_user.id
+      expect(response).to render_template 'edit'
+    end
+
+    it 'handles invalid updates' do
+      allow(@old_user).to receive(:save).and_return(false)
+      post :update, id: @old_user.id, user: { email: 'some email address'}
+      expect(response).to render_template 'edit'
+      new_user = controller.instance_variable_get(:@user)
+      expect(new_user[:email]).to eq 'some email address'
```

Comment:
L49 & 50 are testing that the newly rendered 'edit' form uses the object's new attributes carried from the original form submission. Assignment of the new attributes to the `@user` object is done by Rails as part of the call to `.update_attributes` and not by our controller. This test is, therefore, unnecessary.